### PR TITLE
chore(progress): add Render function to periodically print a Renderer

### PR DESCRIPTION
**Pull request by efekarakus**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/term/cursor/cursor.go [hunk 1]
```diff
@@ -13,23 +14,54 @@ import (
...
 type Cursor struct {
 	c cursor
 }
 
-// New creates a new cursor that writes to stderr and reads from stdin.
+// New creates a new cursor that writes to stderr.
 func New() *Cursor {
 	return &Cursor{
 		c: &terminal.Cursor{
-			In:  os.Stdin,
```
Comment (iamhopaul123) [inline internal/pkg/term/cursor/cursor.go:27]: Why do we remove this one? Is it not used?

File: internal/pkg/term/progress/render.go [hunk 2]
```diff
@@ -14,6 +22,28 @@ type Renderer interface {
...
+		select {
+		case <-ctx.Done():
+			_, err := r.Render(out)
+			return err
+		case <-time.After(renderInterval):
+			nl, err := r.Render(out)
+			if err != nil {
+				return err
+			}
+			cursor.Up(nl + 1) // move the cursor back up to the starting line so that the Renderer is rendered in-place.
```
Comment (iamhopaul123) [inline internal/pkg/term/progress/render.go:42]: Does this work as expected for Linux and Windows?

**Review by iamhopaul123 — APPROVED** [review]
LGTM! the only concern is if it works as expected in the other platform.

File: internal/pkg/term/cursor/cursor.go (hunk 1 above)
Comment (efekarakus) [reply internal/pkg/term/cursor/cursor.go:27]: It's not used yeah 😅

File: internal/pkg/term/progress/render.go (hunk 2 above)
Comment (efekarakus) [reply internal/pkg/term/progress/render.go:42]: I haven't checked it on windows yet! I'm thinking of getting the display working e2e on Mac and Linux first and then readjusting according on windows

**Review by huanjani — APPROVED** [review]
LGTM

**Pull request merged**